boards/sim/sim/sim/scripts/Make.defs

```diff
@@ -71,17 +74,12 @@ ARCHWARNINGSXX = -Wall -Wshadow -Wundef
 # Add -fvisibility=hidden
 # Because we don't want export nuttx's symbols to share libraries
 
-ARCHCPUFLAGS += -fvisibility=hidden
-ARCHCPUFLAGSXX += -fvisibility=hidden
-
-# Add -fno-common because macOS "ld -r" doesn't seem to pick objects
-# for common symbols.
-ARCHCPUFLAGS += -fno-common
-ARCHCPUFLAGSXX += -fno-common
+ARCHCFLAGS += -fvisibility=hidden
```

Comment:
Maybe we can move this above the same as `-fno-common` was moved?
Or keep both `-fno-common` and `-fvisibility=hidden` split

Reply:
This modification is aligned with other Make.defs in PR: https://github.com/apache/incubator-nuttx/pull/5553